### PR TITLE
Bump CLI version, Stack LTS-17.10, MVar -> TVar

### PR DESCRIPTION
**Pull request by expede**

### CODE REVIEW

File: fission-cli/library/Fission/CLI/Handler/App/Publish.hs [hunk 1]
```diff
@@ -145,37 +149,56 @@ handleTreeChanges ::
...
+  -> TVar Text
   -> WatchManager
   -> FilePath -- ^ Build dir
   -> IO StopListening
 handleTreeChanges runner userProof appURL copyFilesFlag timeCache hashCache watchMgr absDir =
-  FS.watchTree watchMgr absDir (\_ -> True) \_ -> runner do
-    now     <- getCurrentTime
-    oldTime <- readMVar timeCache
-
-    unless (diffUTCTime now oldTime < Time.doherty) do
```
Comment (expede) [inline fission-cli/library/Fission/CLI/Handler/App/Publish.hs:158]: This started to bother the Rubyist in me. `unless` leads to a lot of double negatives.

File: fission-cli/library/Fission/CLI/Handler/App/Publish.hs [hunk 2]
```diff
@@ -145,37 +149,56 @@ handleTreeChanges ::
...
+        if diffUTCTime now oldTime > Time.doherty
+          then do
+            writeTVar timeCache now
+            return True
+
+          else
+            return False
+
+      when update do
+        Process.sleepThread Time.dohertyMicroSeconds
```
Comment (expede) [inline fission-cli/library/Fission/CLI/Handler/App/Publish.hs:173]: Wait, why? Did we get confused initially?

File: fission-cli/library/Fission/CLI/Remote.hs [hunk 3]
```diff
@@ -21,12 +21,18 @@ getRemoteBaseUrl :: MonadRemote m => m BaseUrl
...
+    Production     -> fromBase production
+    Staging        -> fromBase staging
+    LocalDev       -> return . showBaseUrl $ BaseUrl Http "localhost" 5001 "ipns"
+    Custom baseUrl -> fromBase baseUrl
+  where
+    fromBase url@BaseUrl{..} =
+      return $ showBaseUrl url
+        { baseUrlHost = "ipfs." <> baseUrlHost
+        , baseUrlPath = "ipns"
+        }
```
Comment (expede) [inline fission-cli/library/Fission/CLI/Remote.hs:35]: This continues to feel brittle, but at now this works with local dev

Comment (expede) [reply fission-cli/library/Fission/CLI/Remote.hs:35]: Oh, wait, we need to make that port 10235(?) for our instance

File: stack.yaml [hunk 4]
```diff
@@ -11,18 +11,15 @@ packages:
...
+  - dimensions-2.1.1.0
   - hfsevents-0.1.6
   - ipfs-1.3.0.3
   - lzma-clib-5.2.2
   - raven-haskell-0.1.4.0
   - rescue-0.4.2.1
-  - servant-auth-0.4.0.0
-  - servant-auth-server-0.4.6.0
-  - servant-auth-swagger-0.2.10.1
-  - servant-swagger-ui-redoc-0.3.3.1.22.3
```
[the remark this answers is not in the document]
Comment (walkah) [reply stack.yaml:25]: where are these going? Or were they just unused?

File: fission-cli/library/Fission/CLI/Handler/App/Publish.hs (hunk 1 above)
Comment (walkah) [reply fission-cli/library/Fission/CLI/Handler/App/Publish.hs:158]: 👍

**Review by walkah — APPROVED** [review]
ship it - Hw8vYF4DNRCKY

🧹 👍

File: stack.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (expede) [reply stack.yaml:25]: They're now available in the Stackage LTS (this PR bumps to 17.10)

**Pull request merged**